From 4b5aab95e34e1cbebc7566c1267cddc2560601c8 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 7 Jul 2017 17:41:43 -0700
Subject: [PATCH 2/2] link with libexecinfo on musl

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 build/config/BUILD.gn | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/config/BUILD.gn b/build/config/BUILD.gn
index 17de20b1e4e9..f579a51d8297 100644
--- a/build/config/BUILD.gn
+++ b/build/config/BUILD.gn
@@ -253,6 +253,7 @@ config("default_libs") {
       "dl",
       "pthread",
       "rt",
+      "execinfo",
     ]
   }
 }
-- 
2.13.2

