From 7cd00cf69e5766ab59760c6d9c4ed89a3fb02628 Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Sun, 5 Feb 2017 08:41:22 +0900
Subject: [PATCH] Set program name from the remoting name

---
 toolkit/xre/nsAppRunner.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
index d3d2d66d5d..3dddf5cdd0 100644
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -3763,7 +3763,7 @@ int XREMain::XRE_mainStartup(bool* aExitFlag) {
 
   // Set program name to the one defined in application.ini.
   {
-    nsAutoCString program(gAppData->name);
+    nsAutoCString program(gAppData->remotingName);
     ToLowerCase(program);
     g_set_prgname(program.get());
   }
