Index: firefox-60.1.0/memory/build/Mutex.h
===================================================================
--- firefox-60.1.0.orig/memory/build/Mutex.h
+++ firefox-60.1.0/memory/build/Mutex.h
@@ -44,7 +44,7 @@ struct Mutex
     if (pthread_mutexattr_init(&attr) != 0) {
       return false;
     }
-    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_ADAPTIVE_NP);
+    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_STALLED);
     if (pthread_mutex_init(&mMutex, &attr) != 0) {
       pthread_mutexattr_destroy(&attr);
       return false;
@@ -107,7 +107,7 @@ typedef Mutex StaticMutex;
 
 #if defined(XP_DARWIN)
 #define STATIC_MUTEX_INIT OS_SPINLOCK_INIT
-#elif defined(XP_LINUX) && !defined(ANDROID)
+#elif defined(XP_LINUX) && !defined(ANDROID) && defined(__GLIBC__)
 #define STATIC_MUTEX_INIT PTHREAD_ADAPTIVE_MUTEX_INITIALIZER_NP
 #else
 #define STATIC_MUTEX_INIT PTHREAD_MUTEX_INITIALIZER
