diff -ur chromium-54.0.2810.2__ORIG/build/common.gypi chromium-54.0.2810.2/build/common.gypi
--- chromium-54.0.2810.2__ORIG/build/common.gypi	2017-03-30 10:05:27.017866621 -0400
+++ chromium-54.0.2810.2/build/common.gypi	2017-03-30 12:33:15.012788454 -0400
@@ -3820,10 +3820,6 @@
                   '-msse2',
                   '-mfpmath=sse',
                   '-mmmx',  # Allows mmintrin.h for MMX intrinsics.
-                  '-m32',
-                ],
-                'ldflags': [
-                  '-m32',
                 ],
                 'conditions': [
                   # Use gold linker for Android ia32 target.
diff -ur chromium-54.0.2810.2__ORIG/build/config/compiler/BUILD.gn chromium-54.0.2810.2/build/config/compiler/BUILD.gn
--- chromium-54.0.2810.2__ORIG/build/config/compiler/BUILD.gn	2017-03-30 10:05:27.017866621 -0400
+++ chromium-54.0.2810.2/build/config/compiler/BUILD.gn	2017-03-30 12:33:15.012788454 -0400
@@ -508,8 +508,6 @@
       ]
       ldflags += [ "-m64" ]
     } else if (current_cpu == "x86") {
-      cflags += [ "-m32" ]
-      ldflags += [ "-m32" ]
       if (!is_nacl) {
         cflags += [
           "-msse2",
diff -ur chromium-54.0.2810.2__ORIG/third_party/icu/icu.gyp chromium-54.0.2810.2/third_party/icu/icu.gyp
--- chromium-54.0.2810.2__ORIG/third_party/icu/icu.gyp	2016-07-28 18:17:07.000000000 -0400
+++ chromium-54.0.2810.2/third_party/icu/icu.gyp	2017-03-30 12:33:15.012788454 -0400
@@ -48,9 +48,6 @@
          target_arch=="mipsel")', {
         'target_conditions': [
           ['_toolset=="host"', {
-            'cflags': [ '-m32' ],
-            'ldflags': [ '-m32' ],
-            'asflags': [ '-32' ],
             'xcode_settings': {
               'ARCHS': [ 'i386' ],
             },
diff -ur chromium-54.0.2810.2__ORIG/v8/gypfiles/toolchain.gypi chromium-54.0.2810.2/v8/gypfiles/toolchain.gypi
--- chromium-54.0.2810.2__ORIG/v8/gypfiles/toolchain.gypi	2017-03-30 10:05:27.017866621 -0400
+++ chromium-54.0.2810.2/v8/gypfiles/toolchain.gypi	2017-03-30 12:33:15.012788454 -0400
@@ -1028,9 +1028,6 @@
                   ['host_arch=="s390" or host_arch=="s390x"', {
                     'cflags': [ '-m31' ],
                     'ldflags': [ '-m31' ]
-                  },{
-                   'cflags': [ '-m32' ],
-                   'ldflags': [ '-m32' ]
                   }],
                 ],
               }],
@@ -1046,9 +1043,6 @@
                   ['host_arch=="s390" or host_arch=="s390x"', {
                     'cflags': [ '-m31' ],
                     'ldflags': [ '-m31' ]
-                  },{
-                   'cflags': [ '-m32' ],
-                   'ldflags': [ '-m32' ],
                   }],
                 ],
               }],
--- chromium-54.0.2810.2__ORIG/v8/src/base/build_config.h	2017-03-30 18:56:01.370627233 -0400
+++ chromium-54.0.2810.2/v8/src/base/build_config.h	2017-03-30 20:03:51.563265532 -0400
@@ -126,9 +126,6 @@
 #endif
 
 // Check for supported combinations of host and target architectures.
-#if V8_TARGET_ARCH_IA32 && !V8_HOST_ARCH_IA32
-#error Target architecture ia32 is only supported on ia32 host
-#endif
 #if (V8_TARGET_ARCH_X64 && V8_TARGET_ARCH_64_BIT && \
      !(V8_HOST_ARCH_X64 && V8_HOST_ARCH_64_BIT))
 #error Target architecture x64 is only supported on x64 host
diff -ur chromium-54.0.2810.2__ORIG/v8/src/base/build_config.h chromium-54.0.2810.2/v8/src/base/build_config.h
--- chromium-54.0.2810.2__ORIG/v8/src/base/build_config.h	2016-07-28 18:17:37.000000000 -0400
+++ chromium-54.0.2810.2/v8/src/base/build_config.h	2017-03-30 12:39:19.929620843 -0400
@@ -137,9 +137,6 @@
      !(V8_HOST_ARCH_X64 && V8_HOST_ARCH_32_BIT))
 #error Target architecture x32 is only supported on x64 host with x32 support
 #endif
-#if (V8_TARGET_ARCH_ARM && !(V8_HOST_ARCH_IA32 || V8_HOST_ARCH_ARM))
-#error Target architecture arm is only supported on arm and ia32 host
-#endif
 #if (V8_TARGET_ARCH_ARM64 && !(V8_HOST_ARCH_X64 || V8_HOST_ARCH_ARM64))
 #error Target architecture arm64 is only supported on arm64 and x64 host
 #endif
