From a761b698240d5b21adf6f69004a76696a9e18b87 Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Sun, 25 Sep 2016 09:22:22 +0900
Subject: [PATCH] Allow to override ICU_DATA_FILE from the environment

---
 build/autoconf/icu.m4 | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/build/autoconf/icu.m4 b/build/autoconf/icu.m4
index 6afa262b07..342e9d30a2 100644
--- a/build/autoconf/icu.m4
+++ b/build/autoconf/icu.m4
@@ -78,7 +78,9 @@ if test -n "$USE_ICU"; then
     # TODO: the l is actually endian-dependent
     # We could make this set as 'l' or 'b' for little or big, respectively,
     # but we'd need to check in a big-endian version of the file.
-    ICU_DATA_FILE="icudt${version}l.dat"
+    if test -z "$ICU_DATA_FILE"; then
+        ICU_DATA_FILE="icudt${version}l.dat"
+    fi
 fi
 
 AC_SUBST(MOZ_ICU_VERSION)
