From d6eb718821bbeecc6d4c558563b9e9e8f4418005 Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Tue, 22 May 2018 07:51:56 +0900
Subject: [PATCH] Avoid using vmrs/vmsr on armel

---
 dom/media/webaudio/blink/DenormalDisabler.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dom/media/webaudio/blink/DenormalDisabler.h b/dom/media/webaudio/blink/DenormalDisabler.h
index b7a3ca321c..4f92d77968 100644
--- a/dom/media/webaudio/blink/DenormalDisabler.h
+++ b/dom/media/webaudio/blink/DenormalDisabler.h
@@ -44,7 +44,7 @@ namespace WebCore {
 #  define HAVE_DENORMAL 1
 #endif
 
-#if defined(__arm__) || defined(__aarch64__)
+#if (defined(__arm__) && !defined(__SOFTFP__)) || defined(__aarch64__)
 #  define HAVE_DENORMAL 1
 #endif
 
