From 74cdba47edc7fe1ac449aca5cfe8720ce38558a9 Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Mon, 3 Sep 2018 07:37:40 +0900
Subject: [PATCH] Use remoting name for call to gdk_set_program_class

Closes: #907574

---
 widget/gtk/nsAppShell.cpp | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/widget/gtk/nsAppShell.cpp b/widget/gtk/nsAppShell.cpp
index b4010bf38d..2af4072ebe 100644
--- a/widget/gtk/nsAppShell.cpp
+++ b/widget/gtk/nsAppShell.cpp
@@ -24,6 +24,7 @@
 #  include "WakeLockListener.h"
 #endif
 #include "gfxPlatform.h"
+#include "nsAppRunner.h"
 #include "ScreenHelperGTK.h"
 #include "HeadlessScreenHelper.h"
 #include "mozilla/widget/ScreenManager.h"
@@ -174,10 +175,8 @@ nsresult nsAppShell::Init() {
     // option when program uses gdk_set_program_class().
     //
     // See https://bugzilla.gnome.org/show_bug.cgi?id=747634
-    nsAutoString brandName;
-    mozilla::widget::WidgetUtils::GetBrandShortName(brandName);
-    if (!brandName.IsEmpty()) {
-      gdk_set_program_class(NS_ConvertUTF16toUTF8(brandName).get());
+    if (gAppData) {
+      gdk_set_program_class(gAppData->remotingName);
     }
   }
 
