From 9ec6007c2615dc661acf69eaca7f4e2a00cdd83f Mon Sep 17 00:00:00 2001
From: Otavio Salvador <otavio@ossystems.com.br>
Date: Wed, 15 Feb 2012 17:33:32 -0200
Subject: [PATCH] Set fullscreen on start

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
---
 src/webconverger.js |   13 +++++++++++++
 1 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/src/webconverger.js b/src/webconverger.js
index 722b288..7e6c2c0 100644
--- a/src/webconverger.js
+++ b/src/webconverger.js
@@ -16,12 +16,25 @@ if (typeof webc == 'undefined') { (function() {
 			}
 		}
 
+                function setFullScreen() {
+                        if (document.documentElement.requestFullScreen) {
+                                document.documentElement.requestFullScreen();
+                        } else if (document.documentElement.mozRequestFullScreen) {
+                                document.documentElement.mozRequestFullScreen();
+                        } else if (document.documentElement.webkitRequestFullScreen) {
+                                document.documentElement.webkitRequestFullScreen(Element.ALLOW_KEYBOARD_INPUT);
+                        }
+                }
+
 		function init() {
 			// Add close tab listener, gBrowser has not been initiated by this point
 			getBrowser().tabContainer.addEventListener("TabClose", tabRemoved, false);
 			getBrowser().tabContainer.addEventListener("DisableDownload", onDownloadStateChange, false);
 			var pbs = Components.classes["@mozilla.org/privatebrowsing;1"].getService(Components.interfaces.nsIPrivateBrowsingService);
 			pbs.privateBrowsingEnabled = true;
+
+                        // Set fullscreen after 500ms
+                        setTimeout(function(){ setFullScreen(); }, 500);
 		}
 
 		webc.init = init();
-- 
1.7.9

