From 77384aa0d93ed6d5f91ada40971f6825e1b9dd69 Mon Sep 17 00:00:00 2001
From: Carlos Rafael Giani <dv@pseudoterminal.org>
Date: Thu, 28 Apr 2016 22:02:39 +0200
Subject: [PATCH] Remove GBM support from wayland.gyp

Signed-off-by: Carlos Rafael Giani <dv@pseudoterminal.org>
---
 ozone/wayland/wayland.gyp | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/wayland/wayland.gyp b/wayland/wayland.gyp
index ff2e210..32f13bc 100644
--- a/ozone/wayland/wayland.gyp
+++ b/ozone/wayland/wayland.gyp
@@ -5,10 +5,6 @@
 
 {
   'variables': {
-    'variables': {
-      'enable_drm_support%': 0,
-    },
-    'enable_drm_support%': '<(enable_drm_support)',
     'conditions': [
       ['sysroot!=""', {
         'pkg-config': '../../build/linux/pkg-config-wrapper "<(sysroot)" "<(target_arch)"',
@@ -42,17 +38,14 @@
       'direct_dependent_settings': {
         'cflags': [
           '<!@(<(pkg-config) --cflags <(wayland_packages))',
-          '<!@(<(pkg-config) --cflags gbm)',
         ],
       },
       'link_settings': {
         'ldflags': [
           '<!@(<(pkg-config) --libs-only-L --libs-only-other <(wayland_packages))',
-          '<!@(<(pkg-config) --libs-only-L --libs-only-other gbm)',
         ],
         'libraries': [
           '<!@(<(pkg-config) --libs-only-l <(wayland_packages))',
-          '<!@(<(pkg-config) --libs-only-l gbm)',
         ],
       },
       'dependencies': [
@@ -65,19 +58,6 @@
         '../..',
         '<(DEPTH)/third_party/khronos',
       ],
-      'conditions': [
-        ['<(enable_drm_support)==1', {
-          'defines': [
-          'ENABLE_DRM_SUPPORT',
-          ],
-          'sources': [
-            'egl/wayland_pixmap.cc',
-            'egl/wayland_pixmap.h',
-            'protocol/wayland-drm-protocol.cc',
-            'protocol/wayland-drm-protocol.h',
-          ],
-        }],
-      ],
       'sources': [
         'data_device.cc',
         'data_device.h',
-- 
2.5.0

