From: Mike Hommey <glandium@debian.org>
Date: Sat, 8 Dec 2007 19:24:40 +0100
Subject: Set javascript.options.showInConsole

---
 modules/libpref/src/init/all.js |    1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/libpref/src/init/all.js b/modules/libpref/src/init/all.js
index e8160c2..3c0a70c 100644
--- a/modules/libpref/src/init/all.js
+++ b/modules/libpref/src/init/all.js
@@ -627,6 +627,7 @@ pref("javascript.options.strict",           false);
 pref("javascript.options.strict.debug",     true);
 #endif
 pref("javascript.options.relimit",          true);
+pref("javascript.options.showInConsole",    true);
 pref("javascript.options.tracejit.content",  false);
 pref("javascript.options.tracejit.chrome",   false);
 pref("javascript.options.methodjit.content", true);
