From e264d70626cddc047967d5a1cab7758c0f584a48 Mon Sep 17 00:00:00 2001
From: Randy MacLeod <Randy.MacLeod@windriver.com>
Date: Mon, 16 Jan 2023 22:11:58 -0500
Subject: [PATCH] drop -gsimple-template-names

Upstream-Status: Inappropriate clang-12 only.
Signed-off-by: Randy MacLeod <Randy.MacLeod@windriver.com>

---
 build/config/compiler/BUILD.gn | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index f8777fe..e2b62b9 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -196,7 +196,10 @@ declare_args() {
   # This greatly reduces the size of debug builds, at the cost of
   # debugging information which is required by some specialized
   # debugging tools.
-  simple_template_names = is_clang && !is_nacl && !is_win && !is_apple
+  #
+  # Disabled because the version of clang used on dunfell doesn't recognize the
+  # argument.
+  simple_template_names = false
 }

 declare_args() {
