From 9ac52b6d0d7a14b35d5054da7022945b5eab61f7 Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Wed, 3 Mar 2021 12:03:36 +0200
Subject: [PATCH] IWYU: add ctime for std::time

Upstream-Status: Backport

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
 ui/accessibility/ax_tree_serializer.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ui/accessibility/ax_tree_serializer.h b/ui/accessibility/ax_tree_serializer.h
index 70a4446682..e1f80f5088 100644
--- a/ui/accessibility/ax_tree_serializer.h
+++ b/ui/accessibility/ax_tree_serializer.h
@@ -8,6 +8,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <ctime>
 #include <ostream>
 #include <unordered_map>
 #include <unordered_set>
