--- a/third_party/skia/src/ports/SkFontHost_FreeType_common.cpp
+++ b/third_party/skia/src/ports/SkFontHost_FreeType_common.cpp
@@ -714,7 +714,11 @@ void colrv1_draw_paint(SkCanvas* canvas,
             canvas->drawPaint(colrPaint);
             break;
         }
+#if FREETYPE_MAJOR == 2 && FREETYPE_MINOR >= 11
         case FT_COLR_PAINTFORMAT_TRANSFORM:
+#else
+        case FT_COLR_PAINTFORMAT_TRANSFORMED:
+#endif
         case FT_COLR_PAINTFORMAT_TRANSLATE:
         case FT_COLR_PAINTFORMAT_SCALE:
         case FT_COLR_PAINTFORMAT_ROTATE:
@@ -771,10 +775,17 @@ void colrv1_transform(FT_Face face,
     SkASSERT(canvas || out_transform);
 
     switch (colrv1_paint.format) {
+#if FREETYPE_MAJOR == 2 && FREETYPE_MINOR >= 11
         case FT_COLR_PAINTFORMAT_TRANSFORM: {
             transform = ToSkMatrix(colrv1_paint.u.transform.affine);
             break;
         }
+#else
+        case FT_COLR_PAINTFORMAT_TRANSFORMED: {
+            transform = ToSkMatrix(colrv1_paint.u.transform.affine);
+            break;
+        }
+#endif
         case FT_COLR_PAINTFORMAT_TRANSLATE: {
             transform = SkMatrix::Translate(
                 SkFixedToScalar(colrv1_paint.u.translate.dx),
@@ -891,11 +902,19 @@ bool colrv1_traverse_paint(SkCanvas* canvas,
             traverse_result = colrv1_start_glyph(canvas, palette, face, paint.u.colr_glyph.glyphID,
                                                  FT_COLOR_NO_ROOT_TRANSFORM);
             break;
+#if FREETYPE_MAJOR == 2 && FREETYPE_MINOR >= 11
         case FT_COLR_PAINTFORMAT_TRANSFORM:
             colrv1_transform(face, paint, canvas);
             traverse_result = colrv1_traverse_paint(canvas, palette, face,
                                                     paint.u.transform.paint, visited_set);
             break;
+#else
+        case FT_COLR_PAINTFORMAT_TRANSFORMED:
+            ccolrv1_transform(face, paint, canvas);
+            traverse_result = colrv1_traverse_paint(canvas, palette, face,
+                                                    paint.u.transform.paint, visited_set);
+            break;
+#endif
         case FT_COLR_PAINTFORMAT_TRANSLATE:
             colrv1_transform(face, paint, canvas);
             traverse_result = colrv1_traverse_paint(canvas, palette, face,
