From e264d70626cddc047967d5a1cab7758c0f584a48 Mon Sep 17 00:00:00 2001
From: Randy MacLeod <Randy.MacLeod@windriver.com>
Date: Mon, 16 Jan 2023 22:11:58 -0500
Subject: [PATCH] drop -gsimple-template-names

Upstream-Status: Inappropriate clang-12 only.
Signed-off-by: Randy MacLeod <Randy.MacLeod@windriver.com>

---
 build/config/compiler/BUILD.gn | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index ff2f5f4fcd482..2f99c80f51371 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -826,9 +826,11 @@ config("compiler") {
   # * Apple platforms (e.g. MacOS, iPhone, iPad) aren't supported because xcode
   #   lldb doesn't have the needed changes yet.
   # TODO(crbug.com/1379070): Remove if the upstream default ever changes.
-  if (is_clang && !is_nacl && !is_win && !is_apple) {
-    cflags_cc += [ "-gsimple-template-names" ]
-  }
+  # Drop for 109 due to dunfell:
+  #   clang-12: error: unknown argument: '-gsimple-template-names'
+  #if (is_clang && !is_nacl && !is_win && !is_apple) {
+  #  cflags_cc += [ "-gsimple-template-names" ]
+  #}

   # MLGO specific flags. These flags enable an ML-based inliner trained on
   # Chrome on Android (arm32) with ThinLTO enabled, optimizing for size.
