diff --git a/content/gpu/BUILD.gn b/content/gpu/BUILD.gn
index 93559f6092c7a..f257fa50077e9 100644
--- a/content/gpu/BUILD.gn
+++ b/content/gpu/BUILD.gn
@@ -141,6 +141,5 @@ target(link_target_type, "gpu_sources") {
   # Use DRI on desktop Linux builds.
   if (current_cpu != "s390x" && current_cpu != "ppc64" && is_linux &&
       !is_castos) {
-    configs += [ "//build/config/linux/dri" ]
   }
 }
diff --git a/media/gpu/sandbox/BUILD.gn b/media/gpu/sandbox/BUILD.gn
index cfcb7fa80ef89..8d149450b2116 100644
--- a/media/gpu/sandbox/BUILD.gn
+++ b/media/gpu/sandbox/BUILD.gn
@@ -33,6 +33,5 @@ source_set("sandbox") {
   if (current_cpu != "s390x" && current_cpu != "ppc64" && is_linux &&
       !is_castos) {
     # For DRI_DRIVER_DIR.
-    configs += [ "//build/config/linux/dri" ]
   }
 }
