Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>
Upstream-Status: Inappropriate [oe specific]

--- a/build/rust/std/BUILD.gn	2026-03-16 07:25:22.000000000 +0100
+++ b/build/rust/std/BUILD.gn	2026-03-16 07:34:45.265759526 +0100
@@ -55,11 +55,7 @@
       "miniz_oxide",
       "object",
     ]
-    if (rustc_nightly_capability) {
-      stdlib_files += [ "adler2" ]
-    } else {
-      stdlib_files += [ "adler" ]
-    }
+    stdlib_files += [ "adler2" ]
   }
 
   if (toolchain_for_rust_host_build_tools) {
