diff --git a/src/webcfullscreen.xul b/src/webcfullscreen.xul
index f2ce99d..5cd3830 100644
--- a/src/webcfullscreen.xul
+++ b/src/webcfullscreen.xul
@@ -82,8 +82,6 @@
 
   <keyset id="mainKeyset">
 
-    <key id="restart" keycode="VK_F10" oncommand="goQuitApplication();"/>
-
     <key id="key_newNavigator" disabled="true"/>
     <key id="key_newNavigatorTab" disabled="true"/>
     <key id="focusURLBar" disabled="true"/>
diff --git a/src/webcnoaddressbar.xul b/src/webcnoaddressbar.xul
index 8bbc322..98c4124 100644
--- a/src/webcnoaddressbar.xul
+++ b/src/webcnoaddressbar.xul
@@ -75,8 +75,6 @@
 
   <keyset id="mainKeyset">
 
-    <key id="restart" keycode="VK_F10" oncommand="goQuitApplication();"/>
-
     <key id="key_newNavigator" disabled="true"/>
     <key id="key_newNavigatorTab" disabled="false"/>
     <key id="focusURLBar" disabled="true"/>
diff --git a/src/webconverger.xul b/src/webconverger.xul
index 33634f3..9b1821a 100644
--- a/src/webconverger.xul
+++ b/src/webconverger.xul
@@ -75,8 +75,6 @@
 
   <keyset id="mainKeyset">
 
-    <key id="restart" keycode="VK_F10" oncommand="goQuitApplication();"/>
-
     <key id="key_newNavigator" disabled="true"/>
     <key id="key_newNavigatorTab" disabled="false"/>
     <key id="focusURLBar" disabled="false"/>
