From c8c4a4290238f9c8e6c71c660815f3cf28646cdf Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Sun, 14 Aug 2022 07:01:33 +0900
Subject: [PATCH] Work around bz#1775202 to fix FTBFS on ppc64el

Upstream-Status: Pending
---
 third_party/libwebrtc/moz.build | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/third_party/libwebrtc/moz.build b/third_party/libwebrtc/moz.build
index aa47ddba57..3651bd817a 100644
--- a/third_party/libwebrtc/moz.build
+++ b/third_party/libwebrtc/moz.build
@@ -600,6 +600,13 @@ if CONFIG["CPU_ARCH"] == "mips64" and CONFIG["OS_TARGET"] == "Linux":
         "/third_party/libwebrtc/third_party/pipewire/pipewire_gn"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DIRS += [
+        "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn",
+        "/third_party/libwebrtc/modules/desktop_capture/primitives_gn",
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
 
     DIRS += [
