From: Mike Hommey <mh@glandium.org>
Date: Sun, 2 Nov 2014 10:26:21 +0900
Subject: Work around binutils assertion on mips

---
 media/libcubeb/tests/moz.build | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/media/libcubeb/tests/moz.build b/media/libcubeb/tests/moz.build
index e98d8d9..c291598 100644
--- a/media/libcubeb/tests/moz.build
+++ b/media/libcubeb/tests/moz.build
@@ -6,16 +6,17 @@
 
 DEFINES['CUBEB_GECKO_BUILD'] = True
 
-GeckoCppUnitTests([
-  'test_tone'
-])
+if CONFIG['CPU_ARCH'] != 'mips':
+    GeckoCppUnitTests([
+      'test_tone'
+    ])
 
-if CONFIG['OS_TARGET'] != 'Android':
-  GeckoCppUnitTests([
-    'test_audio',
-    'test_latency',
-    'test_sanity'
-  ])
+    if CONFIG['OS_TARGET'] != 'Android':
+      GeckoCppUnitTests([
+        'test_audio',
+        'test_latency',
+        'test_sanity'
+      ])
 
 LOCAL_INCLUDES += [
     '../include'
