From ba6d42aff5c66a42fa9e1b685b58bd342adf62a0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@googlemail.com>
Date: Thu, 12 Jan 2017 08:44:26 +0100

---
 old-configure | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/old-configure b/old-configure
index bd7299504..30ec9e111 100644
--- a/old-configure
+++ b/old-configure
@@ -10639,6 +10639,8 @@ fi
 	echo $ac_n "checking for NSS - version >= $min_nss_version""... $ac_c" 1>&6
 echo "configure:10641: checking for NSS - version >= $min_nss_version" >&5
 
+        NSS_CONFIG="pkg-config nss"
+        nss_config_args=
 	no_nss=""
 	if test "$NSS_CONFIG" = "no"; then
 		no_nss="yes"
@@ -10646,11 +10648,11 @@ echo "configure:10641: checking for NSS - version >= $min_nss_version" >&5
 		NSS_CFLAGS=`$NSS_CONFIG $nss_config_args --cflags`
 		NSS_LIBS=`$NSS_CONFIG $nss_config_args --libs`
 
-		nss_config_major_version=`$NSS_CONFIG $nss_config_args --version | \
+		nss_config_major_version=`$NSS_CONFIG $nss_config_args --modversion | \
 			sed 's/\([0-9]*\)\.\([0-9]*\)\(\.\([0-9]*\)\)\{0,1\}/\1/'`
-		nss_config_minor_version=`$NSS_CONFIG $nss_config_args --version | \
+		nss_config_minor_version=`$NSS_CONFIG $nss_config_args --modversion | \
 			sed 's/\([0-9]*\)\.\([0-9]*\)\(\.\([0-9]*\)\)\{0,1\}/\2/'`
-		nss_config_micro_version=`$NSS_CONFIG $nss_config_args --version | \
+		nss_config_micro_version=`$NSS_CONFIG $nss_config_args --modversion | \
 			sed 's/\([0-9]*\)\.\([0-9]*\)\(\.\([0-9]*\)\)\{0,1\}/\4/'`
 		if test -z "$nss_config_micro_version"; then
 			nss_config_micro_version="0"
