diff -Nur firefox-52.1.0esr-bak/old-configure firefox-52.1.0esr/old-configure
--- firefox-52.1.0esr-bak/old-configure	2017-04-18 03:33:03.000000000 +0900
+++ firefox-52.1.0esr/old-configure	2017-05-17 17:14:09.684432744 +0900
@@ -10638,6 +10638,8 @@
 	echo $ac_n "checking for NSS - version >= $min_nss_version""... $ac_c" 1>&6
 echo "configure:10640: checking for NSS - version >= $min_nss_version" >&5
 
+        NSS_CONFIG="pkg-config nss"
+        nss_config_args=
 	no_nss=""
 	if test "$NSS_CONFIG" = "no"; then
 		no_nss="yes"
@@ -10645,11 +10647,11 @@
 		NSS_CFLAGS=`$NSS_CONFIG $nss_config_args --cflags`
 		NSS_LIBS=`$NSS_CONFIG $nss_config_args --libs`
 
-		nss_config_major_version=`$NSS_CONFIG $nss_config_args --version | \
+		nss_config_major_version=`$NSS_CONFIG $nss_config_args --modversion | \
 			sed 's/\([0-9]*\)\.\([0-9]*\)\(\.\([0-9]*\)\)\{0,1\}/\1/'`
-		nss_config_minor_version=`$NSS_CONFIG $nss_config_args --version | \
+		nss_config_minor_version=`$NSS_CONFIG $nss_config_args --modversion | \
 			sed 's/\([0-9]*\)\.\([0-9]*\)\(\.\([0-9]*\)\)\{0,1\}/\2/'`
-		nss_config_micro_version=`$NSS_CONFIG $nss_config_args --version | \
+		nss_config_micro_version=`$NSS_CONFIG $nss_config_args --modversion | \
 			sed 's/\([0-9]*\)\.\([0-9]*\)\(\.\([0-9]*\)\)\{0,1\}/\4/'`
 		if test -z "$nss_config_micro_version"; then
 			nss_config_micro_version="0"
