diff -Nur firefox-52.1.0esr-bak/old-configure firefox-52.1.0esr/old-configure
--- firefox-52.1.0esr-bak/old-configure	2017-04-18 03:33:03.000000000 +0900
+++ firefox-52.1.0esr/old-configure	2017-05-17 17:18:03.209152491 +0900
@@ -10236,11 +10236,14 @@
 	echo $ac_n "checking for NSPR - version >= $min_nspr_version""... $ac_c" 1>&6
 echo "configure:10238: checking for NSPR - version >= $min_nspr_version" >&5
 
+	NSPR_CONFIG="pkg-config nspr"
+	nspr_config_args=
 	no_nspr=""
 	if test "$NSPR_CONFIG" != "no"; then
 		NSPR_CFLAGS=`$NSPR_CONFIG $nspr_config_args --cflags`
 		NSPR_LIBS=`$NSPR_CONFIG $nspr_config_args --libs`
-		NSPR_VERSION_STRING=`$NSPR_CONFIG $nspr_config_args --version`	
+		NSPR_VERSION_STRING=`$NSPR_CONFIG $nspr_config_args --modversion`	
+	    echo -n "$NSPR_VERSION_STRING... "
 	elif test -n "${NO_NSPR_CONFIG_SYSTEM_VERSION}"; then
 	    NSPR_CFLAGS="${NO_NSPR_CONFIG_SYSTEM_CFLAGS}"
 		NSPR_LIBS="${NO_NSPR_CONFIG_SYSTEM_LDFLAGS}"
