diff -Nur firefox-52.1.0esr-bak/config/external/nss/crmf/moz.build firefox-52.1.0esr/config/external/nss/crmf/moz.build
--- firefox-52.1.0esr-bak/config/external/nss/crmf/moz.build	2017-04-11 11:13:09.000000000 +0900
+++ firefox-52.1.0esr/config/external/nss/crmf/moz.build	2017-05-17 18:12:29.157152491 +0900
@@ -8,7 +8,6 @@
 
 if CONFIG['MOZ_SYSTEM_NSS']:
     OS_LIBS += [l for l in CONFIG['NSS_LIBS'] if l.startswith('-L')]
-    OS_LIBS += ['-lcrmf']
 else:
     USE_LIBS += [
         # The dependency on nss is not real, but is required to force the
diff -Nur firefox-52.1.0esr-bak/old-configure firefox-52.1.0esr/old-configure
--- firefox-52.1.0esr-bak/old-configure	2017-04-18 03:33:03.000000000 +0900
+++ firefox-52.1.0esr/old-configure	2017-05-17 18:14:28.369152491 +0900
@@ -10692,12 +10692,6 @@
 
 fi
 
-if test -n "$MOZ_SYSTEM_NSS"; then
-   NSS_LIBS="$NSS_LIBS -lcrmf"
-else
-   NSS_CFLAGS="-I${DIST}/include/nss"
-fi
-
 if test -z "$SKIP_LIBRARY_CHECKS"; then
 # Check whether --with-system-jpeg or --without-system-jpeg was given.
 if test "${with_system_jpeg+set}" = set; then
