From: Mike Hommey <mh@glandium.org>
Date: Mon, 16 Dec 2013 10:53:58 +0900
Subject: Add debian/extra-stuff to upstream build system directory traversal

---
 moz.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/moz.build b/moz.build
index fee89428deea..0eff8b810f05 100644
--- a/moz.build
+++ b/moz.build
@@ -86,4 +86,7 @@ if not CONFIG['JS_STANDALONE'] and CONFIG['MOZ_BUILD_APP']:
 
 CONFIGURE_SUBST_FILES += ['.cargo/config']
 
+if not CONFIG['LIBXUL_SDK']:
+    DIRS += ['debian/extra-stuff']
+
 include('build/templates.mozbuild')
