From: Mike Hommey <glandium@debian.org>
Date: Wed, 12 Mar 2008 23:25:15 +0100
Subject: Don't install system profile

---
 xulrunner/app/Makefile.in |    2 --
 1 file changed, 2 deletions(-)

diff --git a/xulrunner/app/Makefile.in b/xulrunner/app/Makefile.in
index eae02da..2d5b8f4 100644
--- a/xulrunner/app/Makefile.in
+++ b/xulrunner/app/Makefile.in
@@ -48,8 +48,6 @@ VPATH = @srcdir@
 
 include $(DEPTH)/config/autoconf.mk
 
-DIRS = profile
-
 PREF_JS_EXPORTS = $(srcdir)/xulrunner.js
 GARBAGE += $(addprefix $(DIST)/bin/defaults/pref/,xulrunner.js)
 
