From: Mike Hommey <glandium@debian.org>
Date: Wed, 12 Mar 2008 23:20:52 +0100
Subject: Don't build example component

---
 xulrunner/Makefile.in |    1 -
 1 file changed, 1 deletion(-)

diff --git a/xulrunner/Makefile.in b/xulrunner/Makefile.in
index 0af55ea..176ace0 100644
--- a/xulrunner/Makefile.in
+++ b/xulrunner/Makefile.in
@@ -47,7 +47,6 @@ PARALLEL_DIRS = \
   app \
   setup \
   stub \
-  examples \
   $(NULL)
 
 ifeq ($(OS_ARCH),WINNT)
