From 6b1a2f0adfcd737d33bc1f0298b5a75ba704c01e Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Mon, 16 Dec 2013 11:11:13 +0900
Subject: Make upstream build system create debian/installer/Makefile


diff --git a/moz.build b/moz.build
index f0e50b2..50044dc 100644
--- a/moz.build
+++ b/moz.build
@@ -77,4 +77,6 @@ if not CONFIG['JS_STANDALONE'] and CONFIG['MOZ_BUILD_APP']:
 if not CONFIG['LIBXUL_SDK']:
     DIRS += ['debian/extra-stuff']
 
+CONFIGURE_SUBST_FILES += ['debian/installer/Makefile']
+
 include('build/templates.mozbuild')
-- 
cgit v0.10.2
