From cc06f755b91a7f59b920ce691bc973075a3e712c Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Fri, 28 Sep 2012 00:38:53 +0200
Subject: Preprocess appstrings.properties

Closes: #688987
---
 browser/locales/jar.mn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/browser/locales/jar.mn b/browser/locales/jar.mn
index 03a8cdf..ea60605 100644
--- a/browser/locales/jar.mn
+++ b/browser/locales/jar.mn
@@ -152,7 +152,7 @@
     locale/browser-region/region.properties        (%chrome/browser-region/region.properties)
 # the following files are browser-specific overrides
     locale/browser/netError.dtd                (%chrome/overrides/netError.dtd)
-    locale/browser/appstrings.properties       (%chrome/overrides/appstrings.properties)
+*   locale/browser/appstrings.properties       (%chrome/overrides/appstrings.properties)
     locale/browser/downloads/settingsChange.dtd  (%chrome/overrides/settingsChange.dtd)
 % override chrome://global/locale/netError.dtd chrome://browser/locale/netError.dtd
 % override chrome://global/locale/appstrings.properties chrome://browser/locale/appstrings.properties
-- 
cgit v0.11.2


