From: Mike Hommey <mh@glandium.org>
Date: Sun, 2 Nov 2014 10:26:21 +0900
Subject: Work around binutils assertion on mips

---
 media/libcubeb/tests/moz.build | 33 +++++++++++++++++----------------
 1 file changed, 17 insertions(+), 16 deletions(-)

diff --git a/media/libcubeb/tests/moz.build b/media/libcubeb/tests/moz.build
index 1b17c7b1cb23..a292b63bbf38 100644
--- a/media/libcubeb/tests/moz.build
+++ b/media/libcubeb/tests/moz.build
@@ -6,24 +6,25 @@
 
 DEFINES['CUBEB_GECKO_BUILD'] = True
 
-GeckoCppUnitTests([
-  'test_duplex',
-  'test_record',
-  'test_tone',
-  'test_utils'
-])
+if CONFIG['CPU_ARCH'] != 'mips':
+    GeckoCppUnitTests([
+      'test_duplex',
+      'test_record',
+      'test_tone',
+      'test_utils'
+    ])
 
-if CONFIG['MOZ_PULSEAUDIO'] or CONFIG['OS_TARGET'] in ('Darwin', 'WINNT', 'Android'):
-  GeckoCppUnitTests([
-    'test_resampler',
-  ])
+    if CONFIG['MOZ_PULSEAUDIO'] or CONFIG['OS_TARGET'] in ('Darwin', 'WINNT', 'Android'):
+      GeckoCppUnitTests([
+        'test_resampler',
+      ])
 
-if CONFIG['OS_TARGET'] != 'Android':
-  GeckoCppUnitTests([
-    'test_audio',
-    'test_latency',
-    'test_sanity'
-  ])
+    if CONFIG['OS_TARGET'] != 'Android':
+      GeckoCppUnitTests([
+        'test_audio',
+        'test_latency',
+        'test_sanity'
+      ])
 
 LOCAL_INCLUDES += [
     '../include',
