From 48a378ae647cbd6f8466bd0e8dbb04400bb175a8 Mon Sep 17 00:00:00 2001
From: Eduardo Valentin <eduardo.valentin@openpossa.org>
Date: Thu, 24 Jul 2008 16:05:09 -0400
Subject: [PATCH 1/1] Remove -Werror from build

Remove -Werror from build

Signed-off-by: Eduardo Valentin <eduardo.valentin@openpossa.org>
---
 configure |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

Index: mozilla-1.9.1/configure
===================================================================
--- mozilla-1.9.1.orig/configure
+++ mozilla-1.9.1/configure
@@ -5525,7 +5525,7 @@ if test "$GNU_CC"; then
         # Don't allow undefined symbols in libraries
         DSO_LDOPTS="$DSO_LDOPTS -Wl,-z,defs"
     fi
-    WARNINGS_AS_ERRORS='-Werror'
+    WARNINGS_AS_ERRORS=''
     DSO_CFLAGS=''
     DSO_PIC_CFLAGS='-fPIC'
     ASFLAGS="$ASFLAGS -fPIC"
@@ -7104,7 +7104,7 @@ EOF
     IMPLIB='emximp -o'
     FILTER='emxexp -o'
     LDFLAGS='-Zmap'
-    WARNINGS_AS_ERRORS='-Werror'
+    WARNINGS_AS_ERRORS=''
     MOZ_DEBUG_FLAGS="-g -fno-inline"
     MOZ_OPTIMIZE_FLAGS="-O2"
     MOZ_OPTIMIZE_LDFLAGS="-s -Zlinker /EXEPACK:2 -Zlinker /PACKCODE -Zlinker /PACKDATA"
@@ -7283,7 +7283,7 @@ else
 fi
 rm -f conftest*
        fi
-       WARNINGS_AS_ERRORS='-Werror'
+       WARNINGS_AS_ERRORS=''
        MOZ_OPTIMIZE_FLAGS="-xO4"
        MKSHLIB='$(CXX) $(CXXFLAGS) $(DSO_PIC_FLAGS) $(DSO_LDOPTS) -h $@ -o $@'
        MKCSHLIB='$(CC) $(CFLAGS) $(DSO_PIC_FLAGS) -G -z muldefs -h $@ -o $@'
