--- mozilla-release/ipc/chromium/src/base/file_util.cc.orig	2012-05-01 08:07:10.000000000 +0200
+++ mozilla-release/ipc/chromium/src/base/file_util.cc	2012-05-01 08:05:02.000000000 +0200
@@ -8,9 +8,7 @@
 #include <io.h>
 #endif
 #include <stdio.h>
-#if defined(ANDROID)
 #include <unistd.h>
-#endif
 
 #include <fstream>
 
--- mozilla-release/ipc/chromium/src/base/message_pump_libevent.cc.orig	2012-05-01 08:16:55.000000000 +0200
+++ mozilla-release/ipc/chromium/src/base/message_pump_libevent.cc	2012-05-01 08:16:37.000000000 +0200
@@ -6,6 +6,7 @@
 
 #include <errno.h>
 #include <fcntl.h>
+#include <unistd.h>
 
 #include "eintr_wrapper.h"
 #include "base/logging.h"
--- mozilla-release/security/nss/lib/softoken/secmodt.h.orig	2012-05-01 13:13:50.000000000 +0200
+++ mozilla-release/security/nss/lib/softoken/secmodt.h	2012-05-01 13:14:25.000000000 +0200
@@ -337,8 +337,9 @@
 /* default module configuration strings */
 #define SECMOD_SLOT_FLAGS "slotFlags=[RSA,DSA,DH,RC2,RC4,DES,RANDOM,SHA1,MD5,MD2,SSL,TLS,AES,Camellia,SEED,SHA256,SHA512]"
 
-#define SECMOD_MAKE_NSS_FLAGS(fips,slot) \
-"Flags=internal,critical"fips" slotparams=("#slot"={"SECMOD_SLOT_FLAGS"})"
+/* from http://gcc.gnu.org/bugzilla/show_bug.cgi?id=50917 */
+#define SECMOD_MAKE_NSS_FLAGS(fips,slot) \                                      
+"Flags=internal,critical" fips" slotparams=("#slot"={" SECMOD_SLOT_FLAGS"})"
 
 #define SECMOD_INT_NAME "NSS Internal PKCS #11 Module"
 #define SECMOD_INT_FLAGS SECMOD_MAKE_NSS_FLAGS("",1)
--- mozilla-release/ipc/chromium/src/base/time_posix.cc.orig	2012-05-01 13:15:33.000000000 +0200
+++ mozilla-release/ipc/chromium/src/base/time_posix.cc	2012-05-01 12:42:38.000000000 +0200
@@ -14,6 +14,8 @@
 #include <time.h>
 #endif
 
+#include <unistd.h>
+
 #include <limits>
 
 #include "base/basictypes.h"
--- mozilla-release/ipc/chromium/src/base/file_util_linux.cc.orig	2012-05-01 08:39:16.000000000 +0200
+++ mozilla-release/ipc/chromium/src/base/file_util_linux.cc	2012-05-01 08:39:39.000000000 +0200
@@ -5,6 +5,7 @@
 #include "base/file_util.h"
 
 #include <fcntl.h>
+#include <unistd.h>
 
 #include <string>
 #include <vector>
