diff -Nur firefox-52.1.0esr-bak/build/moz.configure/old.configure firefox-52.1.0esr/build/moz.configure/old.configure
--- firefox-52.1.0esr-bak/build/moz.configure/old.configure	2017-04-11 11:13:08.000000000 +0900
+++ firefox-52.1.0esr/build/moz.configure/old.configure	2017-05-17 15:50:24.008097491 +0900
@@ -33,7 +33,7 @@
     for ac in (mozconfig_autoconf, autoconf, 'autoconf-2.13', 'autoconf2.13',
                'autoconf213'):
         if ac:
-            autoconf = find_program(ac)
+            autoconf = find_program(ac, ["/usr/bin"])
             if autoconf:
                 break
     else:
