diff -Nur firefox-51.0b9-bak/build/subconfigure.py firefox-51.0b9/build/subconfigure.py
--- firefox-51.0b9-bak/build/subconfigure.py	2016-09-20 01:19:29.000000000 +0900
+++ firefox-51.0b9/build/subconfigure.py	2016-12-26 15:10:26.621879238 +0900
@@ -161,7 +161,7 @@
     # Scan the config.status output for information about configuration files
     # it generates.
     config_status_output = subprocess.check_output(
-        [data['shell'], '-c', '%s --help' % config_status],
+        [data['shell'], '-c', '%s %s --help' % (sys.executable, config_status)],
         stderr=subprocess.STDOUT).splitlines()
     state = None
     for line in config_status_output:
@@ -372,7 +372,7 @@
             sys.stdout.flush()
         try:
             output += subprocess.check_output([data['shell'], '-c',
-                './config.status'], stderr=subprocess.STDOUT, cwd=objdir,
+                sys.executable, './config.status'], stderr=subprocess.STDOUT, cwd=objdir,
                 env=data['env'])
         except subprocess.CalledProcessError as e:
             ret = e.returncode
