--- /tmp/Linux.mk	2010-03-23 18:52:06.000000000 +0100
+++ mozilla-1.9.2/security/coreconf/Linux.mk	2010-03-23 18:52:57.000000000 +0100
@@ -52,52 +52,8 @@
 
 DEFAULT_COMPILER = gcc
 
-ifeq ($(OS_TEST),ppc64)
-	CPU_ARCH	= ppc
-ifeq ($(USE_64),1)
-	ARCHFLAG	= -m64
-endif
-else
-ifeq ($(OS_TEST),alpha)
-        OS_REL_CFLAGS   = -D_ALPHA_
-	CPU_ARCH	= alpha
-else
-ifeq ($(OS_TEST),x86_64)
-ifeq ($(USE_64),1)
-	CPU_ARCH	= x86_64
-else
-	OS_REL_CFLAGS	= -Di386
-	CPU_ARCH	= x86
-	ARCHFLAG	= -m32
-endif
-else
-ifeq ($(OS_TEST),sparc64)
-	CPU_ARCH        = sparc
-else
-ifeq (,$(filter-out arm% sa110,$(OS_TEST)))
-	CPU_ARCH        = arm
-else
-ifeq (,$(filter-out parisc%,$(OS_TEST)))
-	CPU_ARCH        = hppa
-else
-ifeq (,$(filter-out i%86,$(OS_TEST)))
-	OS_REL_CFLAGS	= -Di386
-	CPU_ARCH	= x86
-else
-ifeq ($(OS_TEST),sh4a)
-	CPU_ARCH        = sh4
-else
-# $(OS_TEST) == m68k, ppc, ia64, sparc, s390, s390x, mips, sh3, sh4
-	CPU_ARCH	= $(OS_TEST)
-endif
-endif
-endif
-endif
-endif
-endif
-endif
-endif
-
+OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+CPU_ARCH =
 
 LIBC_TAG		= _glibc
 
