From: Zack Weinberg <zackw@panix.com>
Date: Fri, 9 Mar 2012 11:07:16 +0100
Subject: Bug 729817 - Allow the Nouveau driver with Mesa >= 8.0.1 - r=bjacob

---
 widget/src/xpwidgets/GfxInfoX11.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/widget/src/xpwidgets/GfxInfoX11.cpp b/widget/src/xpwidgets/GfxInfoX11.cpp
index c6b2ad3..a2d5e0b 100644
--- a/widget/src/xpwidgets/GfxInfoX11.cpp
+++ b/widget/src/xpwidgets/GfxInfoX11.cpp
@@ -314,9 +314,9 @@ GfxInfo::GetFeatureStatusImpl(PRInt32 aFeature,
             *aStatus = nsIGfxInfo::FEATURE_BLOCKED_DRIVER_VERSION;
             aSuggestedDriverVersion.AssignLiteral("Not Mesa");
         }
-        if (mIsNouveau) {
+        if (mIsNouveau && version(mMajorVersion, mMinorVersion) < version(8,0)) {
             *aStatus = nsIGfxInfo::FEATURE_BLOCKED_DRIVER_VERSION;
-            aSuggestedDriverVersion.AssignLiteral("<Not the Nouveau driver>");
+            aSuggestedDriverVersion.AssignLiteral("Mesa 8.0");
         } else if (version(mMajorVersion, mMinorVersion, mRevisionVersion) < version(7,10,3)) {
             *aStatus = nsIGfxInfo::FEATURE_BLOCKED_DRIVER_VERSION;
             aSuggestedDriverVersion.AssignLiteral("Mesa 7.10.3");
