From: Mike Hommey <mh@glandium.org>
Date: Fri, 30 Sep 2011 09:51:16 +0200
Subject: Bug 690682 - Disable dead symbol removal when failing to test for
 bug 670659

---
 configure.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.in b/configure.in
index cf849e1..ba1b528 100644
--- a/configure.in
+++ b/configure.in
@@ -6874,7 +6874,7 @@ if test "$GNU_CC" -a "$GCC_USE_GNU_LD" -a -n "$MOZ_DEBUG_FLAGS"; then
             fi
         else
              dnl We really don't expect to get here, but just in case
-             AC_ERROR([couldn't compile a simple C file])
+             GC_SECTIONS_BREAKS_DEBUG_RANGES="no, but it's broken in some other way"
         fi
         rm -rf conftest*])
     if test "$GC_SECTIONS_BREAKS_DEBUG_RANGES" = no; then
