From fccec0d49843b48e46b1da3ef72de8893d7177fb Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Mon, 21 Nov 2016 08:15:38 +0900
Subject: [PATCH] Work around Debian bug #844357

---
 widget/gtk/mozgtk/gtk3/moz.build | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/widget/gtk/mozgtk/gtk3/moz.build b/widget/gtk/mozgtk/gtk3/moz.build
index 4e9379565d..538ef1acfc 100644
--- a/widget/gtk/mozgtk/gtk3/moz.build
+++ b/widget/gtk/mozgtk/gtk3/moz.build
@@ -36,3 +36,7 @@ OS_LIBS += [
     'gdk-3',
 ]
 OS_LIBS += as_needed
+
+# Work around Debian bug #844357
+if CONFIG['CPU_ARCH'].startswith('mips'):
+    OS_LIBS += ['-Wl,--no-gc-sections']
