From 89d5f9e227acb9237a7b5b90697ce5e41c3e2a89 Mon Sep 17 00:00:00 2001
From: Mario Domenech Goulart <mario@ossystems.com.br>
Date: Wed, 29 Feb 2012 14:21:01 -0300
Subject: [PATCH] Disable C-l

---
 src/webconverger.xul |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/webconverger.xul b/src/webconverger.xul
index 33634f3..7e40f11 100644
--- a/src/webconverger.xul
+++ b/src/webconverger.xul
@@ -79,7 +79,7 @@
 
     <key id="key_newNavigator" disabled="true"/>
     <key id="key_newNavigatorTab" disabled="false"/>
-    <key id="focusURLBar" disabled="false"/>
+    <key id="focusURLBar" disabled="true"/>
     <key id="focusURLBar2" disabled="true"/>
     <key id="key_search" disabled="false"/>
     <key id="key_search2" disabled="false"/>
-- 
1.7.2.5

