From 651f74efd75c06d4160f075362ee8141b4f1227a Mon Sep 17 00:00:00 2001
From: Mike Hommey <glandium@debian.org>
Date: Thu, 13 Mar 2008 20:30:19 +0100
Subject: [PATCH] Set DPI to system settings

---
 modules/libpref/init/all.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/libpref/init/all.js b/modules/libpref/init/all.js
index 9714c8d563..8e5d48b19f 100644
--- a/modules/libpref/init/all.js
+++ b/modules/libpref/init/all.js
@@ -3035,7 +3035,7 @@ pref("layout.css.report_errors", true);
 // A value of 0 means use the system DPI. A positive value is used as the DPI.
 // This sets the physical size of a device pixel and thus controls the
 // interpretation of physical units such as "pt".
-pref("layout.css.dpi", -1);
+pref("layout.css.dpi", 0);
 
 // Set the number of device pixels per CSS pixel. A value <= 0 means choose
 // automatically based on user settings for the platform (e.g., "UI scale factor"
