From: Mike Hommey <mh+mozilla@glandium.org>
Date: Fri, 14 Oct 2011 13:36:04 +0200
Subject: Bug 694533 - LDRH/STRH/LDRSB/STRSB are supported on ARMv5-

---
 js/src/assembler/assembler/ARMAssembler.h |    6 ------
 1 files changed, 0 insertions(+), 6 deletions(-)

diff --git a/js/src/assembler/assembler/ARMAssembler.h b/js/src/assembler/assembler/ARMAssembler.h
index 7a512a1..cd0d540 100644
--- a/js/src/assembler/assembler/ARMAssembler.h
+++ b/js/src/assembler/assembler/ARMAssembler.h
@@ -205,11 +205,9 @@ namespace JSC {
             FCMPD = 0x0eb40b40,
             FSQRTD = 0x0eb10bc0,
             DTR = 0x05000000,
-#if WTF_ARM_ARCH_VERSION >= 5
             LDRH = 0x00100090,
             STRH = 0x00000090,
             DTRH = 0x00000090,
-#endif
             STMDB = 0x09200000,
             LDMIA = 0x08b00000,
             B = 0x0a000000,
@@ -597,8 +595,6 @@ namespace JSC {
                          (posOffset ? DT_UP : 0), rd, rb, offset);
             } else {
                 /* All 16 bit ops and 8 bit unsigned use the newer encoding.*/
-                /*these instructions don't exist before ARMv4*/
-                ASSERT(WTF_ARM_ARCH_VERSION >= 4);
                 emitInst(static_cast<ARMWord>(cc) | DTRH | HDT_IMM | DT_PRE |
                          (isLoad ? DT_LOAD : 0) |
                          (size == 16 ? HDT_UH : 0) |
@@ -637,7 +633,6 @@ namespace JSC {
                          OP2_OFSREG, rd, rb, rm);
             } else {
                 /* All 16 bit ops and 8 bit unsigned use the newer encoding.*/
-                ASSERT(WTF_ARM_ARCH_VERSION >= 4);
                 emitInst(static_cast<ARMWord>(cc) | DTRH | DT_PRE |
                          (isLoad ? DT_LOAD : 0) |
                          (size == 16 ? HDT_UH : 0) |
@@ -737,7 +732,6 @@ namespace JSC {
         // Data transfers like this:
         //  LDRSB rd, [rb, #-offset]
         //  STRSB rd, [rb, #-offset]
-        // TODO: this instruction does not exist on arm v4 and earlier
         void dtrsb_d(bool isLoad, int rd, int rb, ARMWord offset, Condition cc = AL)
         {
             ASSERT(isLoad); /*can only do signed byte loads, not stores*/
