From: Mike Hommey <mh+mozilla@glandium.org>
Date: Sat, 19 Nov 2011 09:37:34 +0100
Subject: Bug 703842 - Avoid R_SPARC_WDISP22 relocation in TrampolineSparc.s

---
 js/src/methodjit/TrampolineSparc.s |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/js/src/methodjit/TrampolineSparc.s b/js/src/methodjit/TrampolineSparc.s
index 0fa9cfd..1b15369 100644
--- a/js/src/methodjit/TrampolineSparc.s
+++ b/js/src/methodjit/TrampolineSparc.s
@@ -112,7 +112,7 @@ throwpoline_exit:
 JaegerInterpolineScripted:
     ld      [%l0 + 0x10], %l0                        /* Load f->prev_ */
     st      %l0, [%fp - 36]                          /* Update f->regs->fp_ */
-    ba     JaegerInterpoline
+    ba     interpoline_enter
     nop
 .size    JaegerInterpolineScripted, . - JaegerInterpolineScripted
 
@@ -120,6 +120,7 @@ JaegerInterpolineScripted:
 .global JaegerInterpoline
 .type   JaegerInterpoline, #function
 JaegerInterpoline:
+interpoline_enter:
     mov     %o0,%o2
     mov     %l3,%o0
     mov     %l2,%o1
