From: Mike Hommey <mh@glandium.org>
Date: Fri, 18 Nov 2011 11:27:29 +0100
Subject: Bug 698923 - Don't require 16 bytes alignment for VMFrame on sparc

---
 js/src/methodjit/MethodJIT.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/src/methodjit/MethodJIT.cpp b/js/src/methodjit/MethodJIT.cpp
index 4f7f5db..c3d01cb 100644
--- a/js/src/methodjit/MethodJIT.cpp
+++ b/js/src/methodjit/MethodJIT.cpp
@@ -257,7 +257,7 @@ JS_STATIC_ASSERT(offsetof(FrameRegs, sp) == 0);
 #if defined(__GNUC__) && !defined(_WIN64)
 
 /* If this assert fails, you need to realign VMFrame to 16 bytes. */
-#ifdef JS_CPU_ARM
+#if defined(JS_CPU_ARM) || defined(JS_CPU_SPARC)
 JS_STATIC_ASSERT(sizeof(VMFrame) % 8 == 0);
 #else
 JS_STATIC_ASSERT(sizeof(VMFrame) % 16 == 0);
