From: Justin Lebar <justin.lebar@gmail.com>
Date: Fri, 27 May 2011 16:50:37 -0400
Subject: Revert bz#164580

---
 xpcom/glue/nsID.h |   14 +++++++++++---
 1 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/xpcom/glue/nsID.h b/xpcom/glue/nsID.h
index 26da5d6..de80a33 100644
--- a/xpcom/glue/nsID.h
+++ b/xpcom/glue/nsID.h
@@ -73,10 +73,18 @@ struct nsID {
    */
 
   inline PRBool Equals(const nsID& other) const {
-    // First cast to void* in order to silence the alignment warnings.
+    // One would think that this could be done faster with a really
+    // efficient implementation of memcmp(), but evidently no
+    // memcmp()'s out there are better than this code.
+    //
+    // See bug http://bugzilla.mozilla.org/show_bug.cgi?id=164580 for
+    // details.
+
     return
-      ((PRUint64*)(void*) &m0)[0] == ((PRUint64*)(void*) &other.m0)[0] &&
-      ((PRUint64*)(void*) &m0)[1] == ((PRUint64*)(void*) &other.m0)[1];
+      ((((PRUint32*) &m0)[0] == ((PRUint32*) &other.m0)[0]) &&
+       (((PRUint32*) &m0)[1] == ((PRUint32*) &other.m0)[1]) &&
+       (((PRUint32*) &m0)[2] == ((PRUint32*) &other.m0)[2]) &&
+       (((PRUint32*) &m0)[3] == ((PRUint32*) &other.m0)[3]));
   }
 
   /**
