--- a/google-chrome	2016-07-08 09:47:11.121810646 -0400
+++ b/google-chrome	2016-07-08 09:47:37.881683637 -0400
@@ -3,6 +3,10 @@
 export CHROME_DEVEL_SANDBOX=/usr/sbin/chrome-devel-sandbox
 export LD_LIBRARY_PATH=/usr/lib/chromium
 
+export GOOGLE_API_KEY="no"
+export GOOGLE_DEFAULT_CLIENT_ID="no"
+export GOOGLE_DEFAULT_CLIENT_SECRET="no"
+
 CHROME_EXTRA_ARGS=""
 if [ "`whoami`" = "root" ] ; then
 	CHROME_EXTRA_ARGS="${CHROME_EXTRA_ARGS} --user-data-dir=ROOT_HOME/.chromium/"
