From 3d64bee51528bd55c4ebb146cd96de3126fd8594 Mon Sep 17 00:00:00 2001
From: Hiroshi Hatake <hatake@clear-code.com>
Date: Mon, 25 Jun 2018 11:41:19 +0900
Subject: [PATCH] Mark GLFeature::framebuffer_multisample as unsupported

Signed-off-by: Takuro Ashie <ashie@clear-code.com>

---
 gfx/gl/GLContext.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/gfx/gl/GLContext.cpp b/gfx/gl/GLContext.cpp
index e2c534a4eb..6388d9fc3c 100644
--- a/gfx/gl/GLContext.cpp
+++ b/gfx/gl/GLContext.cpp
@@ -751,6 +751,16 @@ bool GLContext::InitImpl() {
       MarkUnsupported(GLFeature::framebuffer_multisample);
     }
 
+#ifdef MOZ_WAYLAND
+    if (Vendor() == GLVendor::Imagination &&
+        Renderer() == GLRenderer::SGX544MP) {
+      // PowerVR SGX544MP's EGL does not have valid
+      // glRenderbufferStorageMultisampleANGLE implementation, which breaks
+      // WebGL.
+      MarkUnsupported(GLFeature::framebuffer_multisample);
+    }
+#endif
+
     if (IsMesa()) {
       // DrawElementsInstanced hangs the driver.
       MarkUnsupported(GLFeature::robust_buffer_access_behavior);
