From 2cbaf81a1525540f199d661cef7d8eb7d3560378 Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Fri, 17 Feb 2012 17:47:15 +0100
Subject: [PATCH] Don't auto-disable extensions in system directories

---
 browser/app/profile/firefox.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 04fd609ac9..d735929909 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -62,7 +62,7 @@ pref("extensions.systemAddon.update.enabled", true);
 
 // Disable add-ons that are not installed by the user in all scopes by default.
 // See the SCOPE constants in AddonManager.jsm for values to use here.
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 3);
 // Scopes to scan for changes at startup.
 pref("extensions.startupScanScopes", 0);
 
