From 7c0031bb96238fe9f38a30ae5fb4821e59fb3ebf Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 13 Feb 2019 09:51:14 -0800
Subject: [PATCH] mallopt() is glibc specific API

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 base/allocator/allocator_check.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/allocator/allocator_check.cc b/base/allocator/allocator_check.cc
index 0ac9fa1975..b6338c8285 100644
--- a/base/allocator/allocator_check.cc
+++ b/base/allocator/allocator_check.cc
@@ -11,7 +11,7 @@
 #include "base/allocator/partition_allocator/shim/winheap_stubs_win.h"
 #endif
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#if defined(__GLIBC__) || BUILDFLAG(IS_CHROMEOS)
 #include <malloc.h>
 #endif
 
