Index: chromium-19.0.1049.3/skia/skia.gyp
===================================================================
--- chromium-19.0.1049.3.orig/skia/skia.gyp	2012-05-14 17:15:13.906097364 +0200
+++ chromium-19.0.1049.3/skia/skia.gyp	2012-05-14 17:15:50.082099369 +0200
@@ -1091,7 +1091,7 @@
             '../third_party/skia/src/opts/opts_check_arm.cpp',
           ],
         }],
-        [ 'armv7 == 1 and arm_neon == 0', {
+        [ 'arm_neon == 0', {
           'sources': [
             '../third_party/skia/src/opts/memset.arm.S',
         ],
