Index: chromium-19.0.1049.3/v8/src/platform-linux.cc
===================================================================
--- chromium-19.0.1049.3.orig/v8/src/platform-linux.cc	2012-05-18 13:56:10.405297259 +0200
+++ chromium-19.0.1049.3/v8/src/platform-linux.cc	2012-05-18 13:56:44.693296511 +0200
@@ -188,7 +188,7 @@
 // calling this will return 1.0 and otherwise 0.0.
 static void ArmUsingHardFloatHelper() {
   asm("mov r0, #0");
-#if defined(__VFP_FP__) && !defined(__SOFTFP__)
+#if defined(__ARM_PCS_VFP) && !defined(__SOFTFP__)
   // Load 0x3ff00000 into r1 using instructions available in both ARM
   // and Thumb mode.
   asm("mov r1, #3");
@@ -213,7 +213,7 @@
 #else
   asm("vmov d0, r0, r1");
 #endif  // __thumb__
-#endif  // defined(__VFP_FP__) && !defined(__SOFTFP__)
+#endif  // defined(__ARM_PCS_VFP) && !defined(__SOFTFP__)
   asm("mov r1, #0");
 }
 
