Index: firefox-60.1.0/media/webrtc/trunk/webrtc/system_wrappers/source/cpu_features_linux.c
===================================================================
--- firefox-60.1.0.orig/media/webrtc/trunk/webrtc/system_wrappers/source/cpu_features_linux.c
+++ firefox-60.1.0/media/webrtc/trunk/webrtc/system_wrappers/source/cpu_features_linux.c
@@ -11,7 +11,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <features.h>
-#if __GLIBC_PREREQ(2, 16)
+#if !defined(__GLIBC__)
+#include <sys/auxv.h>
+#elif __GLIBC_PREREQ(2, 16)
 #include <sys/auxv.h>
 #else
 #include <fcntl.h>
@@ -29,7 +31,10 @@ uint64_t WebRtc_GetCPUFeaturesARM(void)
   int architecture = 0;
   unsigned long hwcap = 0;
   const char* platform = NULL;
-#if __GLIBC_PREREQ(2, 16)
+#if !defined(__GLIBC__)
+  hwcap = getauxval(AT_HWCAP);
+  platform = (const char*)getauxval(AT_PLATFORM);
+#elif __GLIBC_PREREQ(2, 16)
   hwcap = getauxval(AT_HWCAP);
   platform = (const char*)getauxval(AT_PLATFORM);
 #else
