From 25354e9557338f815675348bf780e885e0c0620e Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Mon, 20 Aug 2018 10:16:20 +0900
Subject: [PATCH] firefox: Update to 60.0esr :tada:

---
 toolkit/modules/PopupNotifications.jsm | 1 -
 1 file changed, 1 deletion(-)

diff --git a/toolkit/modules/PopupNotifications.jsm b/toolkit/modules/PopupNotifications.jsm
index 6250700774..8627ff591f 100644
--- a/toolkit/modules/PopupNotifications.jsm
+++ b/toolkit/modules/PopupNotifications.jsm
@@ -1541,7 +1541,6 @@ PopupNotifications.prototype = {
         Services.console.logStringMessage("PopupNotifications._onButtonEvent: " +
                                           "Button click happened before the window was focused");
         this.window.focus();
-        return;
       }
 
       let timeSinceShown = this.window.performance.now() - notification.timeShown;
