lss: Match syscalls to match musl

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
--- a/third_party/lss/linux_syscall_support.h
+++ b/third_party/lss/linux_syscall_support.h
@@ -132,15 +132,18 @@ extern "C" {
 
 /* The Android NDK's <sys/stat.h> #defines these macros as aliases
  * to their non-64 counterparts. To avoid naming conflict, remove them. */
-#ifdef __ANDROID__
+#if defined(fstat64)
   /* These are restored by the corresponding #pragma pop_macro near
    * the end of this file. */
 # pragma push_macro("stat64")
 # pragma push_macro("fstat64")
+# pragma push_macro("fstatat64")
 # pragma push_macro("lstat64")
 # undef stat64
 # undef fstat64
+# undef fstatat64
 # undef lstat64
+# define __RESTORE_MACRO 1
 #endif
 
 #if defined(__ANDROID__) && defined(__x86_64__)
@@ -4539,12 +4542,14 @@ struct kernel_statfs {
 # endif
 #endif
 
-#ifdef __ANDROID__
+#ifdef __RESTORE_MACRO
   /* These restore the original values of these macros saved by the
    * corresponding #pragma push_macro near the top of this file. */
 # pragma pop_macro("stat64")
 # pragma pop_macro("fstat64")
+# pragma pop_macro("fstatat64")
 # pragma pop_macro("lstat64")
+#undef __RESTORE_MACRO
 #endif
 
 #if defined(__cplusplus) && !defined(SYS_CPLUSPLUS)
