diff --git a/ui/gl/gl.gyp b/ui/gl/gl.gyp
index 3d0ffb8..d38e87a 100644
--- a/ui/gl/gl.gyp
+++ b/ui/gl/gl.gyp
@@ -36,6 +36,15 @@
           '<(gl_binding_output_dir)',
         ],
       },
+     # Disable narrowing-conversion-in-initialization-list warnings in that we
+     # do not want to fix
+     # ../../ui/gl/gl_image_glx.cc: In member function 'virtual bool gfx::GLImageGLX::Initialize()'
+     # ../../ui/gl/gl_image_glx.cc:90:3: error: narrowing conversion of
+     # 'visinfo.scoped_ptr_malloc<C, FreeProc>::operator-><XVisualInfo,
+     #   gfx::{anonymous}::ScopedPtrXFree>()->XVisualInfo::visualid'
+     # from 'VisualID {aka long unsigned int}' to 'int' inside { } is ill-formed in C++11 [-Werror=narrowing]
+     'cflags+': ['-Wno-narrowing'],
+     'cflags_cc+': ['-Wno-narrowing'],
      'sources': [
         'gl_bindings.h',
         'gl_bindings_skia_in_process.cc',
